src/txn-wal/src/txns.rs

```diff
@@ -727,10 +718,7 @@ where
             .expect("codecs have not changed");
         let (key_schema, val_schema) = match schemas {
             Some((_, key_schema, val_schema)) => (Arc::new(key_schema), Arc::new(val_schema)),
-            None => {
-                tracing::warn!("falling back to default schemas for {}", data_id);
-                (Arc::clone(&self.key_schema), Arc::clone(&self.val_schema))
-            }
+            None => unreachable!("data shard {} should have a schema", data_id),
```

Comment:
Is this right?
- For new shards we will always have at least one schema registered by the time we reach this point, because that happens at txn-registration time.
- For pre-existing shards, every txns shard will have had `open_writer` called on it at least once in the previous release, so the schema should exist.

Reply:
Yup! Gonna copy-paste that as a comment :D